_posts/2022-12-21-easily-verify-your-rust-in-ci-with-kani.md

```diff
@@ -0,0 +1,102 @@
+This assures developers that a code change does not introduce a bug for all the inputs that the test suite exercises.
+For example, on our own Kani repo, a CI status report on a PR might look like this:
+
+<img src="{{site.baseurl | prepend: site.url}}/assets/images/kani-ci-checks.png" alt="A list of tests and checks run on a pull request as part of continuous integration, with all checks required and passing" />
+
+For developers looking for a higher assurance of software quality, Kani proofs can be validated automatically as part of continuous integration.
+To this end, we are excited to announce the [Kani Rust Verifier Action](https://github.com/marketplace/actions/kani-rust-verifier) on the GitHub Marketplace.
+
+## How to use the Kani GitHub action
+
+If you have a [Rust Cargo](https://doc.rust-lang.org/cargo/) project in a GitHub repo, using the Kani CI action is as simple as adding it to your GitHub Actions workflow file.
+Your Kani checks will appear as part of the same workflow as your existing tests, as shown in this [s2n-quic](https://github.com/aws/s2n-quic) CI report:
+
+<img src="{{site.baseurl | prepend: site.url}}/assets/images/kani-in-s2n-quic-ci.png" alt="s2n-quic CI report showing both tests and Kani proofs verified on a pull request" />
+
+To get the latest version, visit the [Kani Verifier action](https://github.com/marketplace/actions/kani-rust-verifier) on the marketplace, and click on the green "Use Latest Version" button, which will give you a yaml snippet you can paste into your `.yml` file.
+<img src="{{site.baseurl | prepend: site.url}}/assets/images/kani-verifier-action.png" alt="A list of tests and checks run on a pull request as part of continuous integration, with all checks required and passing" />
+
+For example, for Kani 0.17, your `kani-ci.yml` file might look like this:
+
+```yaml
+name: Kani CI
+on:
+  pull_request:
+  push:
+jobs:
+  run-kani:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: 'Checkout your code.'
+        uses: actions/checkout@v3
+
+# You can get the latest version from
+# https://github.com/marketplace/actions/kani-rust-verifier
+      - name: 'Run Kani on your code.'
+        uses: model-checking/kani-github-action@v0.17
+```
+
+For more advanced use cases, we provide facilities to override the working directory, as well as to configure the Kani command itself.
+For example, the [s2n-quic](https://github.com/aws/s2n-quic) project uses [the following CI configuration](https://github.com/aws/s2n-quic/blob/main/.github/workflows/ci.yml#L613), which overrides the working directory, and enables the `--tests` option of `cargo kani`.
```

Comment:
This is a nice example.

Reply:
We probably want to link to a hash rather than `main` since this line can change and it would be nice for the blog link to stay correct.

https://github.com/aws/s2n-quic/blob/c221530beb37addf81c7ed58461a945750a1b251/.github/workflows/ci.yml#L613